eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -383,12 +383,18 @@ class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, wallet: OnChainA
       s(e)
   }
 
-  def createLocalParams(nodeParams: NodeParams, initFeatures: Features[InitFeature], channelType: SupportedChannelType, isInitiator: Boolean, fundingAmount: Satoshi)(implicit ec: ExecutionContext): Future[LocalParams] = {
-    if (channelType.paysDirectlyToWallet) {
+  def createLocalParams(nodeParams: NodeParams, initFeatures: Features[InitFeature], channelType: SupportedChannelType, isInitiator: Boolean, fundingAmount: Satoshi)(implicit ec: ExecutionContext): Future[(LocalParams, Alias)] = {
+    val localParams_f = if (channelType.paysDirectlyToWallet) {
       wallet.getReceivePubkey().map(walletKey => makeChannelParams(nodeParams, initFeatures, Script.write(Script.pay2wpkh(walletKey)), Some(walletKey), isInitiator, fundingAmount))
     } else {
       wallet.getReceiveAddress().map(address => makeChannelParams(nodeParams, initFeatures, Script.write(addressToPublicKeyScript(address, nodeParams.chainHash)), None, isInitiator, fundingAmount))
     }
+    val localAlias_p = Promise[Alias]()
+    context.system.eventStream.publish(Register.GenerateLocalAlias(localAlias_p))
```

Comment:
This isn't very academic, but:
- using the `eventStream` saves us from having to pass a reference to `register` all the way to the `peer`
- using a `Promise` saves us from creating an ad-hoc actor (`pipeTo` doesn't work with the `eventStream` approach)